crates/iota-sdk-ffi/src/types/type_tag.rs

```diff
@@ -1,5 +1,179 @@
+
+#[uniffi::export]
+impl TypeTag {
+    #[inline]
+    pub fn is_u8(&self) -> bool {
+        self.0.is_u8()
+    }
+
+    #[inline]
+    pub fn is_u16(&self) -> bool {
+        self.0.is_u16()
+    }
+
+    #[inline]
+    pub fn is_u32(&self) -> bool {
+        self.0.is_u32()
+    }
+
+    #[inline]
+    pub fn is_u64(&self) -> bool {
+        self.0.is_u64()
+    }
+
+    #[inline]
+    pub fn is_u128(&self) -> bool {
+        self.0.is_u128()
+    }
+
+    #[inline]
+    pub fn is_u256(&self) -> bool {
+        self.0.is_u256()
+    }
+
+    #[inline]
+    pub fn is_bool(&self) -> bool {
+        self.0.is_bool()
+    }
+
+    #[inline]
+    pub fn is_address(&self) -> bool {
+        self.0.is_address()
+    }
+
+    #[inline]
+    pub fn is_signer(&self) -> bool {
+        self.0.is_signer()
+    }
+
+    #[inline]
+    pub fn is_vector(&self) -> bool {
+        self.0.is_vector()
+    }
+
+    #[inline]
+    pub fn as_vector_type_tag_opt(&self) -> Option<Arc<TypeTag>> {
+        self.0
+            .as_vector_type_tag_opt()
+            .cloned()
+            .map(Into::into)
+            .map(Arc::new)
+    }
+
+    #[inline]
+    pub fn as_vector_type_tag(&self) -> TypeTag {
+        self.0.as_vector_type_tag().clone().into()
+    }
+
+    #[inline]
+    pub fn is_struct(&self) -> bool {
+        self.0.is_struct()
+    }
+
+    #[inline]
+    pub fn as_struct_tag_opt(&self) -> Option<Arc<super::struct_tag::StructTag>> {
+        self.0
+            .as_struct_tag_opt()
+            .cloned()
+            .map(Into::into)
+            .map(Arc::new)
+    }
+
+    #[inline]
+    pub fn as_struct_tag(&self) -> super::struct_tag::StructTag {
+        self.0.as_struct_tag().clone().into()
+    }
+
+    #[uniffi::constructor]
+    pub fn u8() -> Self {
+        Self(iota_types::TypeTag::U8)
+    }
+
+    #[uniffi::constructor]
+    pub fn u16() -> Self {
+        Self(iota_types::TypeTag::U16)
+    }
+
+    #[uniffi::constructor]
+    pub fn u32() -> Self {
+        Self(iota_types::TypeTag::U32)
+    }
+
+    #[uniffi::constructor]
+    pub fn u64() -> Self {
+        Self(iota_types::TypeTag::U64)
+    }
+
+    #[uniffi::constructor]
+    pub fn u128() -> Self {
+        Self(iota_types::TypeTag::U128)
+    }
+
+    #[uniffi::constructor]
+    pub fn u256() -> Self {
+        Self(iota_types::TypeTag::U256)
+    }
+
+    #[uniffi::constructor]
+    pub fn bool() -> Self {
+        Self(iota_types::TypeTag::Bool)
+    }
+
+    #[uniffi::constructor]
+    pub fn address() -> Self {
+        Self(iota_types::TypeTag::Address)
+    }
+
+    #[uniffi::constructor]
+    pub fn signer() -> Self {
+        Self(iota_types::TypeTag::Signer)
+    }
+
+    #[uniffi::constructor]
+    pub fn vector(type_tag: &TypeTag) -> Self {
+        Self(iota_types::TypeTag::Vector(Box::new(type_tag.0.clone())))
+    }
+
+    #[uniffi::constructor]
+    pub fn struct_tag(struct_tag: &super::struct_tag::StructTag) -> Self {
```

Comment:
Rather than these long paths, can we just use imports?

Reply:
https://github.com/iotaledger/iota-rust-sdk/pull/83/commits/dd42fbb0cff9f95ea705db1c21e180dfe587c4ae